### PR TITLE
updated bootstrap with optional arguments

### PR DESCRIPTION
**Pull request by tinyroy**

### CODE REVIEW

**Comment by aaronpederson** [comment]
What are we attempting to solve with this change?

File: bootstrap.py [hunk 1]
```diff
@@ -258,10 +261,23 @@ def preconfigure():
...
+            continue
         get_dependencies(playbook)
         get_vault(playbook)
         execute(playbook)
 
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--skip-preconfigure', action='store_true', help='Skip configuration')
+parser.add_argument('--skip-core-playbook', action='store_true', help='Skip core playbook');
+parser.add_argument('--skip-base-playbook', action='store_true', help='Skip base playbook');
```
Comment (pdufault) [inline bootstrap.py]: This argument seems to do nothing at all?

**Comment by pdufault** [comment]
##### These options are all with the goal of speeding iteration time when developing ansible playbooks and working with forge.

None of these additions change the default behavior of forge.

##### Skip downloading playbooks from S3

--skip-download allows you to run forge on a given system and develop your playbook locally, rather than having to update the S3 bucket on each edit of your playbook.  This is nice, as you wouldn't have to update S3 with untested playbooks (potentially impacting production environments) while trying something out.

##### Skip installing forge dependencies

--skip-preconfigure skips installing the forge dependencies, which is needless after forge is run the first time and wastes time.

##### Skip running the core playbook

--skip-core-playbook skips running the core playbook, with the goal of saving execution time when testing playbook changes.

##### Skip running the base playbook

--skip-base-playbook does nothing at all yet.  Trivial to complete, I can add that functionality in tomorrow.  This feature would let us run forge on systems specifically to manage user accounts.

All in all, these changes look pretty good to me.
It's not yet ready to merge: we need to ensure that the argparse python library is installed in our base Ubuntu AMI (it isn't usually, nor should we assume it is), and if not:
- catch exceptions around the argparse import and the argument parsing so forge doesn't bail if the argparse library is not installed
- install argparse in preconfigure()

**Comment by aaronpederson** [comment]
Team Approved

**Pull request merged**